actions/ulog_ingestion/src/ulog_ingestion/utils.py

```diff
@@ -0,0 +1,287 @@
+    schema = {"type": "object", "properties": {}, "required": []}
+
+    for field_type, array_size, field_name in message_definition:
+        if field_name.startswith("_padding"):
+            continue
+
+        json_field_type = TYPE_MAPPING.get(field_type, "string")
+
+        if array_size > 1:
+            schema_property = {
+                "type": "array",
+                "items": {"type": json_field_type},
+                "minItems": array_size,
+                "maxItems": array_size,
+            }
+        else:
+            schema_property = {"type": json_field_type}
+
+        schema["properties"][field_name] = schema_property
+        schema["required"].append(field_name)
+
+    return schema
+
+
+def parse_values_to_json(values: List[Tuple[str, str, Any]]) -> Dict[str, Any]:
+    """
+    Parses a list of field values into a JSON-compatible format.
+
+    This function iterates over each tuple in the provided list, handling both
+    array and non-array types. It constructs a JSON object based on field names,
+    types, and values. Array fields are accumulated before being added to the JSON object.
+
+    Args:
+    - values: A list of tuples, each containing a field name, field type, and field value.
+
+    Returns:
+    - A dictionary representing the JSON object constructed from the input values.
+    """
+    json_instance = {}
+    array_values = {}
+
+    for field_name, field_type, field_value in values:
+        if "[" in field_name:
+            array_name = field_name.split("[")[0]
+            array_values.setdefault(array_name, []).append(
+                convert_value(field_type, field_value)
+            )
+        else:
+            json_instance[field_name] = convert_value(field_type, field_value)
+
+    json_instance.update(array_values)
+    return json_instance
+
+
+def convert_value(field_type: str, value: Any) -> Any:
+    """
+    Converts a field value to its corresponding JSON type.
+
+    Args:
+    - field_type: The type of the field as a string.
+    - value: The value to be converted.
+
+    Returns:
+    - The converted value in its appropriate JSON type.
+    """
+    json_type = TYPE_MAPPING.get(field_type, "string")
+    if json_type == "integer":
+        return int(value)
+    elif json_type == "number":
+        return float(value)
+    elif json_type == "boolean":
+        return bool(int(value))
+    elif json_type == "string":
+        return str(value)
+    else:
+        return value
+
+
+def create_message_path_records(topic: Any, field_data: Any) -> None:
+    """
+    Creates message path records for a given topic.
+
+    Args:
+    - topic: The topic object to which the message paths will be added.
+    - field_data: A list of field data objects containing the message definition.
+    """
+    array_list = list()
+    for field in field_data:
+        # For now only allow these types. TODO: Add support for nested types
+        if field.type_str in TYPE_MAPPING_CANONICAL.keys():
```

Comment:
If it's not in this type mapping, we can still fall back to using [`CanonicalType.Uknown`](https://github.com/roboto-ai/roboto-hosted-app/blob/8b608af4745beaba766bbdd7245757bd7cf94f0e/packages/roboto/src/roboto/domain/topics/topic_record.py#L111) so that we can still show that we parsed the message path

Reply:
Yep, I changed this in order to also populate message paths if we don't know the type (using CanonicalType.Unknown). However, we don't yet support .mcap generation for nested (unknown) message types - I'll look into this in another pass.